### DISCUSSION TITLE
Getting Error: async/await is not yet supported in Client Components, only Server Components but hard reload resolves it?

### DISCUSSION THREAD

**Discussion by ridhwaans** [vercel/next.js#71207] [Help]
### Summary

This occurs in App router and next dev (local). Check below for environment information

**Steps to reproduce**
1. Perform a CTA on a nextjs client component page (denoted by `use client`) with this code. The router push routes to a server component page that further renders its own client component
```
`use client`
import { useRouter } from 'next/navigation';
...
const router = useRouter();
...
onClick={() =>
      router.push(`/music/artists/${item.artist.spotifyId}`)
 }
```
2. encounter Unhandled runtime error
```
Unhandled Runtime Error

Error: async/await is not yet supported in Client Components, only Server Components. This error is often caused by accidentally adding `'use client'` to a module that was originally written for the server.
```
3. hard refresh the page (ctrl+r browser)
4. page loads normally

**Current behaviour:**
Getting the error message page

**Expected behaviour:**
Dont display the error message page and show the next page without errors

### Additional information

```js
Environment information
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Wed Feb 19 06:37:35 UTC 2020
  Available memory (MB): 7919
  Available CPU cores: 16
Binaries:
  Node: 20.13.1
  npm: 10.5.2
  Yarn: 1.22.10
  pnpm: 9.1.0
Relevant Packages:
  next: 14.2.3 // There is a newer version (14.2.15) available, upgrade recommended!
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: N/A
Next.js Config:
  output: N/A
```

**Comment by icyJoseph** [comment]
The problem here is that, the router.push call, is calling React stateful hooks, useReducer to be precise. So then, the rendering step, runs into a client component that is using `async/await`, and the client side renderer can't handle that... so it breaks. You can use the `use` function to consume promises, given they keep their reference, on the client, similarly to async/await, https://react.dev/reference/react/use#noun-labs-1201738-(2)

**Reply by JannikZed** [reply]
what I just don't understand - why are the docs showing exactly this thing: async function being triggered on hydration by useEffect to talk to the server?? And exactly this fails for me with the known error?
<img width="663" alt="Screenshot 2024-10-22 at 18 22 55" src="https://github.com/user-attachments/assets/283a8672-7f46-405d-b980-2dbd5a40999d">

**Reply by icyJoseph** [reply]
You probably made the component, not only the effect, asynchronous as well.

Show us the component.

**Reply by JannikZed** [reply]
yes, you are right.. just noticed, that I have a different component nested that was mistakenly using async in the function definition. So the issue was not the data fetching.
